docker/service.go

```diff
@@ -300,14 +278,28 @@ func (s *Service) up(ctx context.Context, imageName string, create bool, options
 		containers = []*Container{c}
 	}
 
-	return s.eachContainer(ctx, containers, func(c *Container) error {
-		if create {
-			if err := s.recreateIfNeeded(ctx, imageName, c, options.NoRecreate, options.ForceRecreate); err != nil {
+	hasBeenRecreate := false
```

Comment:
Should probably name this `hasBeenCreated` in fact.. It's here to not have an infinite loop (with events)